.github/workflows/deploy.yaml

```diff
@@ -119,3 +124,24 @@ jobs:
           publish_dir: ${{ github.repository }}/forrt-website
           publish_branch: gh-pages
           cname: forrt.org
+
+  deploy-test:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    needs: build
+    if: github.event.inputs.pr_number
+    steps:
+      - name: Download Artifact - Website
+        uses: actions/download-artifact@v4
+        with:
+          name: forrt-website-${{ github.run_number }}
+          path: ${{ github.repository }}/forrt-website
+
+      - name: Deploy - GitHub Pages (Test)
```

Comment:
Unless I'm mistaken, this will effectively will push changes to the `gh-pages` branch which will trigger deployment against the 'production' host, even with the CNAME file is changed. The domain used for our pages, and the branch that the website is deployed from, is defined in our repository settings (the latter is an outdated method and needs to be updated).

With regards to GitHub pages, you can have one project site for an organisation and unlimited project (repo) sites. However, GitHub pages does not offer having 'multiple environments' for testing in a single project which is very annoying for a collaborate development lifecycle.

I need to take deeper look - but a workaround would be to have a secondary repository (project site host) with another subdomain (test/staging.forrt.org) that will host a staging environment site and domain. This will need some work that I will explore in short-term, as again, we don't have a way to stage in a 'pre-production' environment.

What is interesting is that Netlify hosting had a staging slot to perform exactly what you are proposing here without workarounds, but it is my understanding that we want to stay on GH Pages for now.

I would advice against merging this for now.

Some SOF posts on the matter:
https://github.com/peaceiris/actions-gh-pages?tab=readme-ov-file#%EF%B8%8F-add-cname-file-cname
https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages
https://stackoverflow.com/questions/59937478/github-pages-staging-site
https://stackoverflow.com/questions/31911091/creating-staging-and-production-versions-of-a-site-on-github-pages

Reply:
The idea is that it deploys to the production environment, to forrt.org/pr-123 - I am not sure if that is fundamentally different to pushing to test.forrt.org, as I don't think that sub-folders can ever interfere with higher level pages on a static webpage? [Agree that the latter would be better, but harder to do.] 

Are you afraid that this would overwrite the production page? I am not 100% sure, but would expect that it would not touch anything outside the destination_dir specified in the action?

Reply:
Fully agree that this is not best practice but I'd have seen it as a major improvement over the status quo while we stay on GitHu. Nevertheless, it should be easy enough to set up deploy to another repo which would be better -I will give that a go when I can.

Reply:
@DAKiersz I now modified the action to push to a staging repository I created - not sure if we need to set up the gh-pages branch there before this works, but the easiest would be to merge and try, given that it no longer touches the production branch - if you are happy with the approach.

Reply:
Speaking from experience - status-quo is magnitude better than interfering with prod. However, as per your updates this should be avoided. I missed that `external_repository` option exists in that action which seems to simplify this process more than expected.

We will need to add the subdomain to our provider and link it to the repository. Myself and @bethaniley work on the this IOI project on Friday's so we can handle to then, and happy to conduct this release for you.